### PR TITLE
[fix] to_dict modifies the underlying message (#151)

### LINKED ISSUE #151: to_dict modifies the underlying message
Hi,

It seems that the `to_dict` method modifies the nested message.

In short, here's the code and the output (version from master, pulled 24-09-2020)

```python
from dataclasses import dataclass
from typing import Dict

import betterproto


class Code(betterproto.Enum):
    ZERO = 0
    ONE = 1

@dataclass(eq=False, repr=False)
class Message(betterproto.Message):
    codes: Dict[str, "Nested"] = betterproto.map_field(2, betterproto.TYPE_STRING, betterproto.TYPE_MESSAGE)

    def __post_init__(self) -> None:
        super().__post_init__()


@dataclass(eq=False, repr=False)
class Nested(betterproto.Message):
    code: "Code" = betterproto.enum_field(1)

    def __post_init__(self) -> None:
        super().__post_init__()

msg = Message(codes={'test': Nested(code=Code.ONE)})
print(msg.codes['test'])
print(msg.to_dict())
print(msg.codes['test'])
```

```
Nested(code=<Code.ONE: 1>)
{'codes': {'test': {'code': 'ONE'}}}
{'code': 'ONE'}
```

Also, if you'll change `ONE` to `ZERO`, it won't print anything, as `ZERO` is a default value (even though it was set manually)

### PR DESCRIPTION
**Pull request by GrownNed**
Fixes #151

### CODE REVIEW

**Comment by Gobot1234** [comment]
Would you mind adding a test for this?

**Comment by Gobot1234** [comment]
I'm not entirely sure what the issue on the tests is but I'd guess the line before the assignment to output should check if output_map (I haven't tested this)

File: tests/test_mapmessage.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from tests.output_betterproto.mapmessage import (
+    Nested,
+    Test,
+)
+
+
+def test_mapmessage():
```
Comment (abn) [inline tests/test_mapmessage.py:7]:
```suggestion
def test_mapmessage_to_dict_preserves_message():
```

Nitpick, but something like that might be good.

**Review by Gobot1234 — APPROVED** [review]

**Comment by Gobot1234** [comment]
Thanks!

**Pull request merged**